packages/core/lib/commands/test/run.js

```diff
@@ -68,7 +68,42 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
+  /**
+   * If test network exists and provider, host or url exists, then it connects to that external network.
+   * If test network exists and provider, host or url doesn't exist, then internal ganache is started with the user specified
+   * configurations in the config.
+   * If no network is specified in the config, then internal ganache is started with default configuration.
+   * If development network exists, then it connects to the development network specified in the config.
```

Comment:
This comment describes 4 conditions, but the code has three. I think the first conditional isn't in the implementation.

Reply:
The first and the last comment can be joined I guess, because that is the 2nd condition in the else if branch. I will change that. Thanks @cds-amal .

Reply:
Nice rework on the refactoring @sukanyaparashar !